### PR TITLE
Added CASLS retainer agreement

### PR DESCRIPTION
**Pull request by h-mcguire**
This PR adds the docx and yml for the CASLS retainer agreement. It also makes necessary adjustments to attachment, objects, and bundles to produce document as output. 

I also removed the attachment block from one of the yml for the plain_langauge_retainer per the Lemma Style Guide.

### CODE REVIEW

**Review by miabonardi — CHANGES_REQUESTED** [review]
Great work here. 

A few things: 

1. Down the CASLS letter path only, after collecting signatures, it collects addresses which are not used on output document. I don't think this should be happening. Maybe try looking into why these are being triggered?

2. The header and logo are overlapping and should be adjusted 
Screen Shot 2024-02-06 at 10 59 24 AM

3. Logo in bottom corner is getting cut off
Screen Shot 2024-02-06 at 11 00 27 AM

**Review by miabonardi — CHANGES_REQUESTED** [review]
Progress here. 

Did you test these changes? 

I ran into an error after selecting CASLS as document: 

Screen Shot 2024-02-20 at 7 55 17 AM

You'll need to change the order block to make the `[i]` be `[0]`. If you need more than 1 client or attorney, you'll want to do a `clients.gather()`

**Comment by h-mcguire** [comment]
I must have totally not tested this - sorry for committing broken code. I was trying to fix the ordering and gather problem (it's gathering the address's for attorney and client after signatures and without me asking for it in any code I wrote). I tried to do it by changing how complete_attribute for attorneys and clients was implemented. I used the [i] based on how the process was discussed in the documentation [here](https://docassemble.org/docs/groups.html#list%20of%20objects).  

I tried to just have complete_attribute defined only within the individual .yml but it didn't fix the gathering issue.

**Comment by miabonardi** [comment]
No worries! 

You're working on an interview order block, and the link you sent is a question block (which is why they are working differently for you). 

Check out how I've edited the example you're looking at to do something similar to what you're going for. You can run this in the playground by creating a new project, and naming the yml file. 

```
objects:
  - location: |
      DAList.using(
        object_type=Address,
        there_are_any=True)
---
# main interview order
# only 1 mandatory True
mandatory: True
code: |
  location.gather()
  download
---
event: download
question: |
  The locations
subquestion: |
  % for loc in location:
  * ${ loc }
  % endfor
---
question: |
  What is the address of the
  ${ ordinal(i) } location?
fields:
  - Address: location[i].address
  - Unit: location[i].unit
    required: False
  - City: location[i].city
  - State: location[i].state
    code: |
      states_list()
  - Zip: location[i].zip
    required: False
---
question: |
  Would you like to add another location?
yesno: location.there_is_another
```

For troubleshooting why its asking for address even when the interview order and the template do not ask for it, on the page it asks for it, select developer information and scroll to the bottom and see why it's being called. A zip code might be required on download in AL, but we can figure it out if this is causing your issue.

**Comment by h-mcguire** [comment]
Thanks! I think it is about the download in AL. It calls CASLS_Retainer_Agreement['final'] and then looks for the clients address set that includes zip. For each variable in the set the interview would look for the object with "_label" after. 

So it looked for clients[0].address.address and pulled up the question block that gathers that, then it looked for clients[0].address.address_label. Then it moved to another variable from the set doing the same thing, ending on clients[0].address.zip_label.  Not sure if this is helpful or makes sense.

**Comment by miabonardi** [comment]
Can you commit the code that was getting this error?

**Comment by h-mcguire** [comment]
Just committed! The address gather problem also happens when you try to do other documents, see the no contact letter for a quick example. 

In the CASLS Retainer there is also a problem where it seems like it is gathering legal_problems before clients[0].name.full despite it being after the name gather. Or more specifically, legal_problems use of ${ clients[0].name } is what triggers the name gather.  If you click the dev mode when it gathers name it makes it seem like legal_problems is what triggers the question - not the main order block.

**Comment by miabonardi** [comment]
@nonprofittechy Is there a way to disable the collection of an address on download or preset it in AL?

**Comment by nonprofittechy** [comment]
Yes, look for the store_variable_snapshot line that is calling for users[0].address.zip

**Comment by miabonardi** [comment]
That's what I thought it was too, but couldn't find it when I looked. Will try again

**Comment by nonprofittechy** [comment]
> That's what I thought it was too, but couldn't find it when I looked. Will try again

You can also use the icon in the playground to browse through mentions of any reference to users[0].address.address, and of course try the developer information button `</>` in the browser when you're running the interview.

image

**Comment by miabonardi** [comment]
I did that for `store_variable_snapshot` and found nothing. For `users[0].address.address` and `users[0].address.zip` only mentions are coming in from `ql_baseline.yml`. I wonder if something in the [al_visual file](docassemble/GBLSLetters/data/questions/al_visual.yml) could be causing this? Should this file be in this repo?

**Comment by tobyfey** [comment]
With GBLS, I also had users[0].address.address being called when it wasn't in the interview or the templates.  It would be called during document production, so I assumed it was in the documents, and later realized it wasn't (after I added a code block that set it to GBLS main office).

File: setup.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def find_package_data(where='.', package='', exclude=standard_exclude, exclude_d
 setup(name='docassemble.GBLSLetters',
       version='0.0.1',
       description=('A docassemble extension.'),
-      long_description='# docassemble.GBLSLetters\r\n\r\nA docassemble extension.\r\n\r\n## Author\r\n\r\nStavros Constantinou, sconsta@bu.edu\r\n\r\n',
+      long_description='# docassemble.GBLS\r\n\r\nLetter Suite for Greater Boston Legal Services.\r\n\r\n## Author\r\n\r\nStavros Constantinou, sconsta@bu.edu\r\n\r\n',
```
Comment (nonprofittechy) [inline setup.py:49]:
```suggestion
      long_description='# docassemble.GBLSLetters\r\n\r\nLetter Suite for Greater Boston Legal Services.\r\n\r\n## Author\r\n\r\nStavros Constantinou, sconsta@bu.edu\r\n\r\n',
```

**Pull request merged**